code/modules/mob/living/silicon/robot/drone/drone_items.dm

```diff
@@ -71,98 +82,68 @@
 	..()
 	can_hold = typecacheof(can_hold)
 
-/obj/item/gripper/attack_self(mob/user as mob)
-	if(wrapped)
-		wrapped.attack_self(user)
-
-/obj/item/gripper/verb/drop_item()
-
-	set name = "Drop Gripped Item"
-	set desc = "Release an item from your magnetic gripper."
-	set category = "Drone"
-
-	drop_item_p()
-
-// The "p" stands for proc, since I was having annoying weird stuff happening with this in the verb
-// when trying to have default values for arguments and stuff
-/obj/item/gripper/proc/drop_item_p(var/silent = 0)
-
-	if(!wrapped)
-		//There's some weirdness with items being lost inside the arm. Trying to fix all cases. ~Z
-		for(var/obj/item/thing in src.contents)
-			thing.forceMove(get_turf(src))
-		return
-
-	if(wrapped.loc != src)
-		wrapped = null
-		return
+/obj/item/gripper/attack_self(mob/user)
+	if(gripped_item)
+		if(gripped_item.type in configurable_items)
+			var/choice = alert(user, "Do you want to configure/place the item, or drop it?", "Gripper selection", "Configure/place", "Drop", "Cancel")
+			switch(choice)
+				if("Configure/place")
+					gripped_item.attack_self(user)
+				if("Drop")
+					drop_gripped_item()
+				if("Cancel")
+					return FALSE
+		else
+			drop_gripped_item()
+	else
+		to_chat(user, "<span class='warning'>[src] is empty.</span>")
 
-	if(!silent)
-		to_chat(src.loc, "<span class='warning'>You drop \the [wrapped].</span>")
-	wrapped.forceMove(get_turf(src))
-	wrapped = null
+/obj/item/gripper/proc/drop_gripped_item(silent = FALSE)
+	if(gripped_item)
+		if(!silent)
+			to_chat(usr, "<span class='warning'>You drop [gripped_item].</span>")
```

Comment:
Changed this from `to_chat(loc` to `to_chat(usr`, because I think using `loc` means it would display the message to all mobs at the location. I'm not sure if using `usr` is the best here. It's simply the easiest solution though, and it works.

Reply:
This will mean that attacking a borg to make him drop his gripped item will give the attacker the message since usr will be the attacker. Best to add user as param

Reply:
I tried adding `mob/user` to params and then doing `to_chat(user` but the message would not show up. Which means every call of this this proc needs to pass a variable to use for user. Am I missing something?

Reply:
Correct. Other procs using user as param also have the same mechanism. It's because (in this case) you will be the usr if you attack the borg and cause it to drop it's gripped item.
You can probably hack the user together in the proc but that won't be an elegant solution. Best will be to pass the real user as a param

Reply:
I think `loc` would always return the borg, no? Unless it's on the floor, which should never happen

Reply:
Actually yeah, for some reason I though loc was giving turf location but that's not the case.

Reply:
That's probably why it used `loc` in the first place, heh

Reply:
Loc in this case was the module seeing the gripper puts it in its contents or something odd